src/LinQuadOptInterface.jl

```diff
@@ -110,55 +110,64 @@ This function updates all of the references in `m`
 after we have deleted row `row` in the affine constraint matrix.
 """
 function shift_references_after_delete_affine!(m, row)
-    for scalar_affine in [
-            cmap(m).less_than,
-            cmap(m).greater_than,
-            cmap(m).equal_to,
-            cmap(m).interval
-        ]
-        for (key, val) in scalar_affine
-            if val > row
-                scalar_affine[key] -= 1
-            end
-        end
-    end
 
-    for vector_affine in [
-            cmap(m).nonnegatives,
-            cmap(m).nonpositives,
-            cmap(m).zeros,
-            cmap(m).vv_nonnegatives,
-            cmap(m).vv_nonpositives,
-            cmap(m).vv_zeros
-        ]
-        for (key, vals) in vector_affine
-            for (i, val) in enumerate(vals)
-                if val > row
-                    vector_affine[key][i] -= 1
-                end
+    cmap_m = cmap(m)
+    _shift_references_after_delete_scalar!(cmap_m.less_than, row)
+    _shift_references_after_delete_scalar!(cmap_m.greater_than, row)
+    _shift_references_after_delete_scalar!(cmap_m.equal_to, row)
+    _shift_references_after_delete_scalar!(cmap_m.interval, row)
+
+
+    _shift_references_after_delete_vector!(cmap_m.nonnegatives, row)
+    _shift_references_after_delete_vector!(cmap_m.nonpositives, row)
+    _shift_references_after_delete_vector!(cmap_m.zeros, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_nonnegatives, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_nonpositives, row)
+    _shift_references_after_delete_vector!(cmap_m.vv_zeros, row)
+
+end
+
+
+# TODO:  This has been added to mirror functionality that was merged into
+# Julia/master in  PR #31223 it will likely be in release v1.2.
+# This code should be removed once support is droped for anything less than v1.2
+if !(hasmethod(Base.map!,Tuple{Any,Base.ValueIterator{<:Dict}}))
+    function Base.map!(f, iter::Base.ValueIterator{<:Dict})
```

Comment:
Instead of overloading `Base.map!`, why not just implement `_future_map!` and leave a TODO.